From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Tue, 10 Jun 2025 12:44:00 +0800
Subject: remoteproc: core: Allow auto retry of rproc_start()

Sometimes the remoteproc just does not want to start for the first time.
---
 drivers/remoteproc/remoteproc_core.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/remoteproc/remoteproc_core.c b/drivers/remoteproc/remoteproc_core.c
index 111111111111..222222222222 100644
--- a/drivers/remoteproc/remoteproc_core.c
+++ b/drivers/remoteproc/remoteproc_core.c
@@ -1432,9 +1432,13 @@ static int rproc_fw_boot(struct rproc *rproc, const struct firmware *fw)
 	}
 
 	ret = rproc_start(rproc, fw);
-	if (ret)
-		goto clean_up_resources;
-
+	if (ret) {
+		// HACK: try again
+		ret = rproc_start(rproc, fw);
+		if (ret) {
+			goto clean_up_resources;
+		}
+	}
 	return 0;
 
 clean_up_resources:
-- 
Armbian

